### PR TITLE
chore: refactor project structure

### LINKED ISSUE #143: Introduce a `prelude`
Instead of exposing everything through the crate's root, we should add a `prelude` instead.

```rs
pub mod prelude;
```

Then importing everything you might need can be done through `use php_parser_rs::prelude::*` instead.

### PR DESCRIPTION
**Pull request by azjezz**
closes #143 

- add prelude
- moved `ast` to `parser`
- moved `Lexer` from `crate::lexer::lexer::Lexer` to `crate::lexer::Lexer`
- renamed `LexerError` to `SyntaxError`
- moved `SyntaxError` to `token::error`
- implemented `Display` for `SyntaxError`
- add `tokens.txt` to integration tests
- removed all unit tests for `Lexer` ( everything is tested in integration tests )
- improved syntax error assertations

### CODE REVIEW

File: tests/0133/code.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+// while PHP allows this, we shouldn't, it's a syntax error that most parsers for other langauges,
```
Comment (ryangjchandler) [inline tests/0133/code.php:3]:
```suggestion
// while PHP allows this, we shouldn't, it's a syntax error that most parsers for other languages,
```

**Review by ryangjchandler — CHANGES_REQUESTED** [review]
Tiny spelling mistake - otherwise perfect. Thanks for removing that lexer config crap, been meaning to get rid of it.

**Comment by azjezz** [comment]
there's a bug here, some parse errors are now lex errors, that doesn't look right.

File: tests/0133/code.php (hunk 1 above)
Comment (azjezz) [reply tests/0133/code.php]: fixed

**Comment by azjezz** [comment]
i believe that `ast` should be moved within `parser`, it doesn't make sense to have `ast` in root, while `token` is in `lexer`.

**Pull request closed without merging**